### ISSUE TITLE
Deletion of pulp deb repo fails

### ISSUE THREAD

**Issue by csomm** [pulp/pulpcore#4208]
**Version**
Katello version 4.7.0

Version of pulp packages:
python39-pulp-container-2.14.3-1.el8.noarch
rubygem-smart_proxy_pulp-3.2.0-3.fm3_3.el8.noarch
rubygem-pulp_certguard_client-1.5.5-1.el8.noarch
python39-pulp-rpm-3.18.5-1.el8.noarch
python39-pulp-ansible-0.15.0-1.el8.noarch
rubygem-pulp_python_client-3.7.3-1.el8.noarch
python39-pulpcore-3.21.0-1.el8.noarch
python39-pulp-deb-2.20.0-1.el8.noarch
rubygem-pulp_deb_client-2.20.0-1.el8.noarch
rubygem-pulp_rpm_client-3.18.7-1.el8.noarch
pulpcore-selinux-1.3.2-1.el8.x86_64
python39-pulp-cli-deb-0.0.2-4.el8.noarch
python39-pulp-cli-0.14.0-4.el8.noarch
python39-pulp-file-1.11.1-1.el8.noarch
python39-pulp-python-3.7.2-2.el8.noarch
python39-pulp-certguard-1.5.5-1.el8.noarch
rubygem-pulp_ansible_client-0.15.0-1.el8.noarch
rubygem-pulp_file_client-1.11.2-1.el8.noarch
rubygem-pulpcore_client-3.21.2-1.el8.noarch
rubygem-pulp_ostree_client-2.0.0-0.1.a1.el8.noarch
rubygem-pulp_container_client-2.14.2-1.el8.noarch

**Describe the bug**
The deletion of a pulp deb repo fails with thee following error:
```
# pulp --force deb repository destroy --name Ubuntu_22_04-1488601
Started background task /pulp/api/v3/tasks/3f52d6e9-4ff3-4db3-bea1-fb9c63e9ab79/
........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Error: Task /pulp/api/v3/tasks/3f52d6e9-4ff3-4db3-bea1-fb9c63e9ab79/ failed: 'FEHLER:  invalid memory alloc request size 1073741824'
```

**To Reproduce**
I'm not sure how to reproduce the problem, but this repo was initially created with distribution "jammy" and arch "amd64", then later we added "jammy-updates" and "Jammy-security" and arch "i386".

This lead to problems with the orphan cleanup, described here:  https://github.com/pulp/pulp_deb/issues/690
The workaround mentioned in the ticket didn't help much to reclaim space, so we wanted to delete and re-create the repo. 

**Expected behavior**
The error should not happen and the repo should be deleted.

**Additional context**
This is an Foreman installation, which is used in production. Because of the broken orphan cleanup we are unable to reclaim space, and we are also unable to delete the repo which seems to cause the orphan cleanup problem. So now the storage requirements for the repos grow each day, without anything we could do about it.
It would be much appreciated, if we can get help on how to delete the repo, so we don't have to re-create the whole Foreman server.

**Comment by dkliban** [comment]
Can you please share the traceback from the failed Pulp task?

**Comment by quba42** [comment]
What happens when you try to delete the repo in question via Katello, rather than using Pulp CLI? It is very dangerous to alter Pulp without going through Katello since Katello won't know why the repo has disappeared.

Also, are you still experiencing problems with orphan cleanup or does orphan cleanup succeed (but not free up sufficient space)?

**Comment by quba42** [comment]
@csomm Thanks, that contains a lot of new information. I think this should be sufficient for a initial investigation.

**Comment by quba42** [comment]
@csomm Having read over this again, I have had the following thought, rather than deleting the repository using `deb repository destroy`, try deleting one repository version at a time? (`deb repository version destroy` starting with the newest version)

Maybe that will result in a more manageable action memory and DB wise.

**Comment by csomm** [comment]
@quba42 Thanks a lot for the hint! 

I thought I already checked the repository versions, to try to delete them, and only found the most recent (108). But it seems I made a mistake, and actually nearly all 108 repository versions were still there. I was able to delete the versions and subsequently delete the repository.

I was also able to do an orphan cleanup with the workaround described in https://github.com/pulp/pulp_deb/issues/690#issuecomment-1346794819 . I think when I tried the workaround earlier, I forgot to restart all pulpcore worker, so this was probably the reason it didn't work. :facepalm: 

Now the orphan cleanup was successful and was able to free up about 200 GB of space.

Thanks again for the help!

**Comment by quba42** [comment]
@csomm It is possible that once you successfully deleted the repo with content blocking the orphan cleanup you no longer needed the orphan cleanup workaround. Anyway I am glad this got resolved.

**Comment by csomm** [comment]
@quba42 I had rolled back the orphan cleanup workaround some time ago, after it didn't work for me. So after I now successfully deleted the repo, I tried the orphan cleanup again, but it still failed. Then I reapplied the orphan cleanup workaround, restarted all pulp services, and then the orphan cleanup worked, and freed up a lot of space.

I also wondered, why I still got the orphan cleanup errors after deleting the repo (and before I reapplied the workaround). I then discovered that there were still other old repos left over, which probably caused the same problems and also deleted them, and did another orphan cleanup, which freed up some more space.

I will probably try to remove the workaround again, to see if we have any more repos, that are causing orphan cleanup problems. I just found https://github.com/pulp/pulp_deb/issues/865#issuecomment-1688585647 , so I will try this to find the repos, in case there are any.

But for now (and until we can do a Foreman/Katello update), I think we can work with the current situation.
